CMakeLists.txt

```diff
@@ -69,6 +70,7 @@ if (WITH_CUDA)
   if (BUILD_SHARED_LIBS)
     set(CUDA_RUNTIME_LIBRARY "Shared")
   else()
+    set(HDF5_USE_STATIC_LIBRARIES True)
```

Comment:
I think you should delete this line in the future. There is no reason we should always demand a static HDF5 if AMS is static. By deleting this you can always explicitly define it in the CLI and have better control. I prefer when possible explicit control to the user instead of implicitly defining things.